packages/hulypulse-client/src/client.ts

```diff
@@ -0,0 +1,311 @@
+  }
+
+  // unsubscribe(key, callback)
+  // return: Promise.resolve if success (true - unsubscribed now, false - already unsubscribed), Promise.reject if error
+  public async unsubscribe(key: string, callback: Callback): Promise<boolean> {
+    if (!this.subscribes[key] || !this.subscribes[key].includes(callback)) {
+      return false
+    }
+    this.subscribes[key] = this.subscribes[key].filter(cb => cb !== callback)
+    if (this.subscribes[key].length === 0) {
+      delete this.subscribes[key]
+      const reply = await this.send({ type: "unsub", key })
+      if (reply.error) {
+        this.subscribes[key] = this.subscribes[key] || []
+        this.subscribes[key].push(callback)
+        throw new Error(reply.error)
+      }
+    }
+    return true
+  }
+
+  // put(key, value, TTL?)
+  // return: Promise.resolve(true) if success
+  public async put(key: string, data: string, TTL?: number): Promise<boolean> {
+    try {
+      const reply = await this.send({ type: "put", key, data, ...(TTL !== undefined ? { TTL } : {}) })
+      if (reply.error) {
+        throw new Error(reply.error)
+      }
+      return Promise.resolve(true)
+    } catch (err) {
+      return Promise.reject(""+err)
+    }
+  }
+
+  // get(key)
+  // return: Promise.resolve(value) if success, Promise.resolve(false) if not found, Promise.reject if error
+  public async get(key: string): Promise<any> {
+    try {
+      const reply = await this.send({ type: "get", key })
+      if (reply.error) {
+        if(reply.error=="not found") {
+          return Promise.resolve(false)
+        }
+        return Promise.reject(new Error(reply.error))
+      }
+      return reply.result?.data
+    } catch (err) {
+      return Promise.reject(""+err)
+    }
+  }
+
+  // get_full(key)
+  // return: Promise.resolve({data, etag, expires_at}) if success, Promise.resolve(false) if not found, Promise.reject if error
+  public async get_full(key: string): Promise<any> {
+    try {
+      const reply = await this.send({ type: "get", key })
+      if (reply.error) {
+        if(reply.error=="not found") {
+          return Promise.resolve(false)
+        }
+        return Promise.reject(new Error(reply.error))
+      }
+      return { data: reply.result.data, etag: reply.result.etag, expires_at: reply.result.expires_at }
+    } catch (err) {
+      return Promise.reject(""+err)
+    }
+  }
+
+  // delete(key)
+  // return: Promise.resolve if success (true - deleted now, false - not found), Promise.reject if error
+  public async delete(key: string): Promise<boolean> {
+    try {
+      const reply = await this.send({ type: "delete", key })
+      if (reply.error) {
+        if(reply.error=="not found") {
+          return Promise.resolve(false)
+        }
+        throw new Error(reply.error)
+      }
+      return Promise.resolve(true)
+    } catch (err) {
+      return Promise.reject("" + err)
+    }
+  }
+
+  // send({}): Promise<any>
+  public async send(arr: any): Promise<any> {
```

Comment:
1. Make private unless we want to expose internal API to users
2. Use proper type instead of any in both places